libs/openchallenges/ui/src/lib/paginator/paginator.component.ts

```diff
@@ -9,22 +18,28 @@ import { PaginatorModule as PrimengPaginatorModule } from 'primeng/paginator';
   templateUrl: './paginator.component.html',
   styleUrls: ['./paginator.component.scss'],
 })
-export class PaginatorComponent implements OnInit {
+export class PaginatorComponent {
   @Input({ required: true }) pageNumber = 0; // index of the new page
   @Input({ required: false }) pageLinkSize = 5;
   @Input({ required: true }) pageSize = 0; // number of rows to display in new page
   @Input({ required: true }) totalRecords = 0;
   @Input({ required: false }) itemsPerPage!: number[];
-
+  @ViewChild('paginator', { static: false }) paginator!: Paginator;
   @Output() pageChange = new EventEmitter();
 
   itemIndex = 0; // index of the first item in the new page
 
-  ngOnInit(): void {
-    this.itemIndex = this.pageNumber * this.pageSize;
-  }
+  // change itemIndex's value will not dynamically trigger selection updates - seems a primeng issue
+  // ngOnInit(): void {
+  //   this.itemIndex = this.pageNumber * this.pageSize;
+  // }
 
   onPageChange(event: any): void {
     this.pageChange.emit(event);
   }
+
+  // Reset paginator to first page
+  resetPage(): void {
```

Comment:
Use a more descriptive name? "resetting the page" does not sound correct. Maybe `resetPageNumber` or `reset` as in reset the paginator.